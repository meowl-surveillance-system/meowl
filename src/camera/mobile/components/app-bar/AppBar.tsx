import React from 'react';
import { View } from 'react-native';
import { Icon } from 'react-native-elements';

/**
 * The AppBar component controls all the main features of MeowlMobile
 */
class AppBar extends React.Component<AppBarProps> {
  constructor(props: AppBarProps) {
    super(props);
  }

  /**
   * Updates the main state of outer components
   * @param props - Property to be updated
   */
  updateProps(props: object) {
    this.props.updateProps(props);
  }

  /**
   * Renders the AppBar component
   *
   * The AppBar contains isPublishing button, settingFormVisible button,
   * flashEnabled button and video-switch button
   */
  render() {
    return (
      <View
        style={{
          flex: 1,
          flexDirection: 'row',
          justifyContent: 'space-evenly',
          backgroundColor: 'navy',
        }}>
        {this.props.isPublishing ? (
          <Icon
            type="material-community"
            name="video-off"
            onPress={() => this.updateProps({ isPublishing: false })}
            reverse={true}
          />
        ) : (
            <Icon
              type="material-community"
              name="video"
              style={{ backgroundColor: 'transparent' }}
              onPress={() => this.updateProps({ isPublishing: true })}
              reverse={true}
            />
          )}
        <Icon
          type="material-community"
          name="settings"
          onPress={() => this.updateProps({ settingsFormVisible: true })}
          reverse={true}
          disabled={this.props.isPublishing}
        />
        <Icon
          type="material-community"
          name="video-switch"
          onPress={() => {
            this.updateProps({
              flashEnabled: false,
              isViewingFrontCamera: !this.props.isViewingFrontCamera,
            });
          }}
          reverse={true}
        />
        {this.props.flashEnabled ? (
          <Icon
            type="material-community"
            name="flash-off"
            onPress={() => this.updateProps({ flashEnabled: false })}
            reverse={true}
            underlayColor="transparent"
            disabled={this.props.isViewingFrontCamera}
          />
        ) : (
            <Icon
              type="material-community"
              name="flash"
              onPress={() => this.updateProps({ flashEnabled: true })}
              reverse={true}
              disabled={this.props.isViewingFrontCamera}
            />
          )}
      </View>
    );
  }
}

export default AppBar;
