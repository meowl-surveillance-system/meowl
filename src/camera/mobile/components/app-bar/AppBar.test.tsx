import 'react-native';
import React from 'react';
import AppBar from './AppBar';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const container = renderer.create(
    <AppBar
      flashEnabled={false}
      isPublishing={false}
      isViewingFrontCamera={false}
      settingsFormVisible={false}
      updateProps={(obj) => { }}
    />
  ).toJSON();
  expect(container).toMatchSnapshot();
});
