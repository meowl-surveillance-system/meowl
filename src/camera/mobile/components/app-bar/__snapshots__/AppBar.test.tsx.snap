// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "navy",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
